Build warning on Windows

The warning (building with `stack`) is:
~~~
ansi-terminal> src\System\Console\ANSI\Windows\Emulator.hs:482:7: warning: [-Wincomplete-uni-patterns]
ansi-terminal>     Pattern match(es) are non-exhaustive
ansi-terminal>     In a pattern binding:
ansi-terminal>         Patterns of type ‘INPUT_RECORD_EVENT’ not matched:
ansi-terminal>             InputMouseEvent _
ansi-terminal>             InputWindowBufferSizeEvent _
ansi-terminal>             InputMenuEvent _
ansi-terminal>             InputFocusEvent _
ansi-terminal>     |
ansi-terminal> 482 |       InputKeyEvent keyEventRecord = inputEvent e
ansi-terminal>     |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
~~~
The underlying code is fine, but it relies on knowing something the compiler cannot know (that a certain input event type implies an input key event). The code could be changed to check that what 'must' be the case is, in fact, the case. I'll make a pull request.

--- a/src/System/Console/ANSI/Windows/Emulator.hs
+++ b/src/System/Console/ANSI/Windows/Emulator.hs
@@ -474,14 +474,17 @@ getReportedCursorPosition
           return $ stringFromInputEvents es
     stringFromInputEvents = cWcharsToChars . wCharsFromInputEvents
     wCharsFromInputEvents = mapMaybe wCharFromInputEvent
-    wCharFromInputEvent e = if isKeyDownEvent
+    wCharFromInputEvent e = if isKeyEvent && isKeyDown
       then Just (unicodeAsciiChar $ keyEventChar keyEventRecord)
       else Nothing
      where
       eventType = inputEventType e
-      InputKeyEvent keyEventRecord = inputEvent e
+      eventRecord = inputEvent e
+      isKeyEvent = eventType == 1
+      keyEventRecord = case eventRecord of
+        InputKeyEvent keyEventRecord' -> keyEventRecord'
+        _ -> error "Unexpected input event, given input event type."
       isKeyDown = keyEventKeyDown keyEventRecord
-      isKeyDownEvent = eventType == 1 && isKeyDown
 
 -- hGetCursorPosition :: Handle -> IO (Maybe (Int, Int))
 -- (See Common-Include.hs for Haddock documentation)

